java-symbol-solver-model/src/main/java/me/tomassetti/symbolsolver/model/typesystem/ReferenceTypeUsage.java

```diff
@@ -292,7 +292,8 @@ protected boolean compareConsideringTypeParameters(ReferenceTypeUsage other) {
         }
         if (this.getQualifiedName().equals(other.getQualifiedName())) {
             if (this.parameters().size() != other.parameters().size()) {
-                throw new IllegalStateException();
+                // one have the type not specified so they are compatible
+                return true;
```

Comment:
Well, I could merge it as it is (removing the last commit possibly) and then we can work on that until all method calls are solved. Once we have done that we can clean up the code which is... not as great as it could be :D